### DISCUSSION TITLE
using cosine similarity with araBERT

### DISCUSSION THREAD

**Discussion by Aml-Hassan-Abd-El-hamid** [aub-mind/arabert#181] [Q&A] [answered]
Hi,
I wanted to ask what is the right way to use cosine similarity with araBERT? I was trying to build a simple semantic similarity search using the "aubmindlab/bert-base-arabertv2" model and I am not sure of my process so if you can check it out and tell me if I can do it in a better way or not, I will be really grateful :)
I used the tokenizer.encode(sentence1) then passed the output of that function to the model() function and then did the same for the other sentence, and then passed the output of the model()  function to the cosine similarity function that returned a tensor that I passed to torch.mean(). 
you can see my code and some test results of it in the figures below :
![Screenshot from 2022-11-11 01-31-27](https://user-images.githubusercontent.com/66205928/201234806-dd984dc6-c6fd-4179-bfb3-8c1837145fe6.png)

![Screenshot from 2022-11-11 01-46-43](https://user-images.githubusercontent.com/66205928/201229630-a640115c-c38c-46bf-95c2-4d2039a40f0c.png)
![Screenshot from 2022-11-11 01-33-31](https://user-images.githubusercontent.com/66205928/201230994-86a1de8d-be68-4249-893c-f6c66ee06bc6.png)
![Screenshot from 2022-11-11 01-33-40](https://user-images.githubusercontent.com/66205928/201231017-05c3b477-decc-4da3-adb2-0f41e5a1661e.png)

**Comment by WissamAntoun** [comment, marked as answer]
Hey, here's a colab with a simple similarity search implemented https://colab.research.google.com/drive/1qECwsKlvip-oXKhjoE0aM1PmuSMwvjdB?usp=sharing

**Reply by Aml-Hassan-Abd-El-hamid** [reply]
Thank you very much for your help, and a special thank you for giving beginners like me the opportunity to work on  Arabic NLP, I've been always looking for such a good tool to work with the Arabic language.
